.github/workflows/compare_latest_version.py

```
@@ -0,0 +1,102 @@
+    ).group(1)
+    # Join the relative URL with the base URL to form an absolute URL
+    full_code_url = urljoin(SNAPHU_URL, latest_version_tar_name)
+
+    return latest_version, full_code_url
+
+
+def get_current_version_from_github_readme() -> str:
+    """
+    Get the current version of the code from the README file in the given GitHub repository.
+
+
+    Returns
+    -------
+    str
+        The current version of the code as mentioned in the README, or an empty string if not found.
+
+    """
+    # Convert the GitHub repo URL to the raw README URL
+    readme_file = Path(__file__).parent.parent.parent / "README"
+    readme = readme_file.read_text()
+
+    # Use regular expression to find version patterns like 'vX.X.X' or 'X.X.X'
+    for line in readme.splitlines():
+        if version_match := re.match(r"Version\s+(\d+\.\d+\.\d+)", line):
+            return version_match.group(1)
+    else:
+        raise ValueError(f"Failed to parse version from readme: {readme}")
+
+
+if __name__ == "__main__":
+    # Find and print the latest version URL
+    remote_version, code_url = fetch_stanford_version_url()
+    current_version = get_current_version_from_github_readme()
+
+    try:
+        if remote_version != current_version:
+            print("::set-output name=new_version_available::true")
+            print("::set-output name=remote_version::{remote_version}")
+            print("New version available: need to update local code...")
+            with open(os.environ["GITHUB_OUTPUT"], "a") as fh:
+                print("new_version_available=true", file=fh)
+                print(f"remove_version={remote_version}", file=fh)
+        else:
+            print("Local version matches remote: nothing to do")
+            with open(os.environ["GITHUB_OUTPUT"], "a") as fh:
+                print("new_version_available=false", file=fh)
+    except KeyError:
+        # Not in github action context
+        pass
```

Let's raise an exception here to the let user know that they invoked this script in a way that wasn't intended.

```suggestion
    except KeyError as e:
        errmsg = (
            "'GITHUB_OUTPUT' must be defined in the environment. Running outside of"
            " GitHub Actions context is unsupported."
        )
        raise RuntimeError(errmsg) from e
```